tests/integration/pure_cairo_vm/cheatcodes/mock_call/mock_call_test.cairo

```diff
@@ -0,0 +1,95 @@
+from starkware.cairo.common.math import assert_not_zero
+
+func test_mock_call_simple() {
+    alloc_locals;
+
+    %{
+        addr = deploy_contract("./src/basic.cairo").contract_address
+        assert addr != 0
+        mock_call(addr, "get_balance", [200])
+        result = call(addr, "get_balance")
+        assert result == [200]
+    %}
+    return ();
+}
+
+func test_mock_call_with_stop() {
+    alloc_locals;
+
+    %{
+        addr = deploy_contract("./src/basic.cairo").contract_address
+        assert addr != 0
+        stop_mock = mock_call(addr, "get_balance", [200])
+        result = call(addr, "get_balance")
+        assert result == [200]
+        stop_mock()
+        result = call(addr, "get_balance")
+        assert result == [100]
+    %}
+    return ();
+}
+
+func test_mock_call_different_addresses() {
+    alloc_locals;
+
+    %{
+        addr1 = deploy_contract("./src/basic.cairo").contract_address
+        assert addr1 != 0
+
+        addr2 = deploy_contract("./src/basic.cairo").contract_address
+        assert addr2 != 0
+
+        stop_mock1 = mock_call(addr1, "get_balance", [200])
+        stop_mock2 = mock_call(addr2, "get_balance", [300])
+
+        result = call(addr1, "get_balance")
+        assert result == [200]
+
+        result = call(addr2, "get_balance")
+        assert result == [300]
+
+        stop_mock1()
+
+        result = call(addr1, "get_balance")
+        assert result == [100]
+
+        result = call(addr2, "get_balance")
+        assert result == [300]
+
+        stop_mock2()
+
+        result = call(addr1, "get_balance")
+        assert result == [100]
+
+        result = call(addr2, "get_balance")
+        assert result == [100]
+
+
+    %}
+    return ();
+}
+
+func test_mock_call_mock_twice_fail() {
+    alloc_locals;
+
+    %{
+        addr = deploy_contract("./src/basic.cairo").contract_address
+        assert addr != 0
+        stop_mock = mock_call(addr, "get_balance", [200])
+        stop_mock = mock_call(addr, "get_balance", [300])
```

Comment:
perhaps we should use this opportunity to have this working? i mean - having a few calls to `mock_call` would work as mocking a few calls in a row, with different parameters. let's discuss this with the team on slack.

Reply:
We can move this discussion to slack or #1251. Here, let's provide a basic version, we can modify it later.